File: trip-kit-booking/src/main/java/com/skedgo/android/tripkit/booking/CompanyInfo.java
package com.skedgo.android.tripkit.booking;

import androidx.annotation.Nullable;

import com.google.gson.annotations.JsonAdapter;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

import static org.immutables.gson.Gson.TypeAdapters;
import static org.immutables.value.Value.Immutable;
import static org.immutables.value.Value.Style;

@TypeAdapters
@Immutable
@Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(GsonAdaptersCompanyInfo.class)
public interface CompanyInfo {
  @Nullable String name();
  @Nullable String website();
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/CompanyInfo.kt
package com.skedgo.tripkit.booking

import com.google.gson.annotations.JsonAdapter
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@TypeAdapters
@Immutable
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersCompanyInfo::class
)
interface CompanyInfo {
    fun name(): String?

    fun website(): String?
}